vnext/ReactUWP/ReactUWP.vcxproj

```diff
@@ -115,54 +115,30 @@
     </Midl>
   </ItemDefinitionGroup>
   <Import Project="$(ReactNativeWindowsDir)\PropertySheets\ReactCommunity.cpp.props" />
-  <ItemDefinitionGroup Condition="'$(Platform)'=='x64' AND '$(OSS_RN)'!='true'">
+  <ItemDefinitionGroup Condition="'$(Platform)'=='x64' AND '$(PATCH_RN)'!='true'">
     <Link>
       <ModuleDefinitionFile>Endpoints\dll\react-native-uwp.x64.def</ModuleDefinitionFile>
     </Link>
-    <ClCompile>
-      <RuntimeTypeInfo>false</RuntimeTypeInfo>
-    </ClCompile>
-  </ItemDefinitionGroup>
-  <ItemDefinitionGroup Condition="'$(Platform)'=='x64' AND '$(OSS_RN)'=='true' AND '$(PATCH_RN)'!='true'">
-    <Link>
-      <ModuleDefinitionFile>Endpoints\dll\react-native-uwp.oss.x64.def</ModuleDefinitionFile>
-    </Link>
   </ItemDefinitionGroup>
-  <ItemDefinitionGroup Condition="'$(Platform)'=='x64' AND '$(OSS_RN)'=='true' AND '$(PATCH_RN)'=='true'">
+  <ItemDefinitionGroup Condition="'$(Platform)'=='x64' AND '$(PATCH_RN)'=='true'">
     <Link>
-      <ModuleDefinitionFile>Endpoints\dll\react-native-uwp.oss.patched.x64.def</ModuleDefinitionFile>
+      <ModuleDefinitionFile>Endpoints\dll\react-native-uwp.patched.x64.def</ModuleDefinitionFile>
     </Link>
   </ItemDefinitionGroup>
-  <ItemDefinitionGroup Condition="('$(Platform)'=='Win32' OR '$(Platform)'=='x86') AND '$(OSS_RN)'!='true'">
+  <ItemDefinitionGroup Condition="('$(Platform)'=='Win32' OR '$(Platform)'=='x86') AND '$(PATCH_RN)'!='true'">
     <Link>
       <ModuleDefinitionFile>Endpoints\dll\react-native-uwp.x86.def</ModuleDefinitionFile>
     </Link>
-    <ClCompile>
-      <RuntimeTypeInfo>false</RuntimeTypeInfo>
-    </ClCompile>
   </ItemDefinitionGroup>
-  <ItemDefinitionGroup Condition="('$(Platform)'=='Win32' OR '$(Platform)'=='x86') AND '$(OSS_RN)'=='true' AND '$(PATCH_RN)'!='true'">
+  <ItemDefinitionGroup Condition="('$(Platform)'=='Win32' OR '$(Platform)'=='x86') AND '$(PATCH_RN)'=='true'">
     <Link>
-      <ModuleDefinitionFile>Endpoints\dll\react-native-uwp.oss.x86.def</ModuleDefinitionFile>
+      <ModuleDefinitionFile>Endpoints\dll\react-native-uwp.patched.x86.def</ModuleDefinitionFile>
     </Link>
   </ItemDefinitionGroup>
-  <ItemDefinitionGroup Condition="('$(Platform)'=='Win32' OR '$(Platform)'=='x86') AND '$(OSS_RN)'=='true' AND '$(PATCH_RN)'=='true'">
-    <Link>
-      <ModuleDefinitionFile>Endpoints\dll\react-native-uwp.oss.patched.x86.def</ModuleDefinitionFile>
-    </Link>
-  </ItemDefinitionGroup>
-  <ItemDefinitionGroup Condition="'$(Platform)'=='ARM' AND '$(OSS_RN)'!='true'">
+  <ItemDefinitionGroup Condition="'$(Platform)'=='ARM'">
     <Link>
       <ModuleDefinitionFile>Endpoints\dll\react-native-uwp.arm.def</ModuleDefinitionFile>
     </Link>
-    <ClCompile>
-      <RuntimeTypeInfo>false</RuntimeTypeInfo>
-    </ClCompile>
```

Comment:
I'm a little skeptical about this.
To be clear: I would like us to build WITHOUT RTTI.
However from what I can see, Folly uses RTTI/dynamic_cast; if we are using Folly through source or a static library (which I think is what we do), then I think we need to have RTTI enabled (RTTI is enabled/disabled on a DLL level)

Reply:
(this *might* be ok if the things they're dynamic_cast-ing are never our types that we derive from theirs)

---
In reply to: [371554430](https://github.com/microsoft/react-native-windows/pull/3948#discussion_r371554430) [](ancestors = 371554430)

Reply:
Folly looks to avoid RTTI unless FOLLY_HAS_RTTI is defined, or portability.h (which we define out) detects it's enabled by vendor specific preprocessor definitions.

Reply:
I did a quick check and yeah, from what I can tell everywhere using a dynamic_cast is test code of experimental. typeid seems to be guarded in what I saw. If we do end up seeing folly issues we should report them since it seems like they try to guard against that definition.